From bcf7abea65b3f50c3c9ca008e6cfaa91ed9d8510 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sun, 7 Jul 2019 15:08:07 -0400
Subject: ISL 0.20 add includes to graphite.h

NOTE: This patch needs work; I have up on fixing it any more.
The PBB_NB_SCATTERING_TRANSFORM change errors out.

And, remove two older includes.
Also, redefine PBB_NB_SCATTERING_TRANSFORM
---
 gcc/graphite.h | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/gcc/graphite.h b/gcc/graphite.h
index cc621b66680..d9136f634b5 100644
--- a/gcc/graphite.h
+++ b/gcc/graphite.h
@@ -24,21 +24,27 @@ along with GCC; see the file COPYING3.  If not see
 
 #include "sese.h"
 #include <isl/options.h>
+#include <isl/ctx.h>
 #include <isl/val_gmp.h>
 #include <isl/set.h>
 #include <isl/union_set.h>
 #include <isl/map.h>
 #include <isl/union_map.h>
-#include <isl/band.h>
 #include <isl/aff.h>
 #include <isl/constraint.h>
 #include <isl/flow.h>
 #include <isl/ilp.h>
 #include <isl/schedule.h>
 #include <isl/ast_build.h>
-#include <isl/val.h>
 
-#ifndef HAVE_ISL_OPTIONS_SET_SCHEDULE_SERIALIZE_SCCS
+#ifdef HAVE_ISL_OPTIONS_SET_SCHEDULE_SERIALIZE_SCCS
+/* isl 0.15 or later.  */
+#include <isl/schedule_node.h>
+#include <isl/id.h>
+#include <isl/space.h>
+
+#else
+/* isl 0.14 or 0.13.  */
 # define isl_stat int
 # define isl_stat_ok 0
 #endif
@@ -390,7 +396,7 @@ struct poly_bb
    out dimension of transformed to count the scatterting transform dimension.
    */
 #define PBB_NB_LOCAL_VARIABLES(PBB) (0)
-#define PBB_NB_SCATTERING_TRANSFORM(PBB) (isl_map_n_out (PBB->transformed))
+#define PBB_NB_SCATTERING_TRANSFORM(PBB) (((PBB)->transformed) ? (((PBB)->transformed)->dim->n_out) : 0)
 #define PBB_IS_REDUCTION(PBB) (PBB->is_reduction)
 
 extern poly_bb_p new_poly_bb (scop_p, void *);
-- 
